signal/signal.go

```diff
@@ -99,6 +103,8 @@ func (c *Interceptor) mainInterruptHandler() {
 
 		case <-c.shutdownRequestChannel:
 			log.Infof("Received shutdown request.")
+			// Adding exit code 1 as this will quit after graceful shutdown
+			c.setExitCode(1)
```

Comment:
We need to always send an exit code into the exit code chan if callers are going to use `GetExitCode`. If we don't have a value to read, it'll block the caller.